adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -645,20 +645,70 @@ private AuthenticationResult acquireTokenSilentSync(final String resource,
                                                         final String apiEventString)
             throws AuthenticationException, InterruptedException {
         final String methodName = ":acquireTokenSilentSync";
+        return acquireTokenSilentSync(null, null, resource, clientId, userId, forceRefresh, claims, apiEventString);
+    }
 
+    /**
+     *
+     * This function tries to acquire token silently. It will first look at the cache
+     * and automatically checks for the token expiration. Additionally, if no suitable
+     * access token is found in the cache, but refresh token is available, the function
+     * will use the refresh token automatically. If both RT and AT are not present, then
+     * it will use the provided assertion and its type to acquire the same.
+     * This method will not show UI for the user. If prompt is needed, the method
+     * will return an exception
+     *
+     * @param assertion the actual saml assertion
+     * @param assertionType version of saml assertion being used
+     * @param resource required resource identifier.
+     * @param clientId required client identifier.
+     * @param userId   UserID obtained from
+     *                 {@link AuthenticationResult #getUserInfo()}
+     * @param forceRefresh when true, access token is renewed using broker if available; otherwise, uses local refresh token
+     * @return A {@link Future} object representing the
+     * {@link AuthenticationResult} of the call. It contains Access
+     * Token,the Access Token's expiration time, Refresh token, and
+     * {@link UserInfo}.
+     * @throws AuthenticationException If silent request fails to get the token back.
+     * @throws InterruptedException    If the main thread is interrupted before or during the activity.
+     */
+    public AuthenticationResult acquireTokenSilentSyncWithAssertion(final String assertion,
+                                                                    final String assertionType,
+                                                                    final String resource,
+                                                                    final String clientId,
+                                                                    final String userId,
+                                                                    final boolean forceRefresh,
+                                                                    final String claims)
+            throws AuthenticationException, InterruptedException {
+        final String methodName = ":acquireTokenSilentSyncWithAssertion";
+        return acquireTokenSilentSync(assertion, assertionType, resource, clientId, userId,
+                forceRefresh, claims, EventStrings.ACQUIRE_TOKEN_WITH_SAML_ASSERTION);
+    }
+
+    private AuthenticationResult acquireTokenSilentSync(final String assertion,
```

Comment:
Can we add nullability annotations here too? `@NonNull` if the param _cannot_ be `null`, `@Nullable` if it can.